ql/pricingengines/vanilla/fdblackscholesvanillaengine.hpp

```diff
@@ -45,21 +45,25 @@ namespace QuantLib {
 
     class FdBlackScholesVanillaEngine : public DividendVanillaOption::engine {
       public:
+        enum CashDividendModel { Spot, Escrowed };
+
         // Constructor
-          FdBlackScholesVanillaEngine(
-                const ext::shared_ptr<GeneralizedBlackScholesProcess>&,
-                Size tGrid = 100, Size xGrid = 100, Size dampingSteps = 0,
-                const FdmSchemeDesc& schemeDesc = FdmSchemeDesc::Douglas(),
-                bool localVol = false,
-                Real illegalLocalVolOverwrite = -Null<Real>());
-
-          FdBlackScholesVanillaEngine(
-                const ext::shared_ptr<GeneralizedBlackScholesProcess>&,
-                const ext::shared_ptr<FdmQuantoHelper>& quantoHelper,
-                Size tGrid = 100, Size xGrid = 100, Size dampingSteps = 0,
-                const FdmSchemeDesc& schemeDesc = FdmSchemeDesc::Douglas(),
-                bool localVol = false,
-                Real illegalLocalVolOverwrite = -Null<Real>());
+        FdBlackScholesVanillaEngine(
+            const ext::shared_ptr<GeneralizedBlackScholesProcess>&,
+            Size tGrid = 100, Size xGrid = 100, Size dampingSteps = 0,
+            const FdmSchemeDesc& schemeDesc = FdmSchemeDesc::Douglas(),
+            bool localVol = false,
+            Real illegalLocalVolOverwrite = -Null<Real>(),
+            CashDividendModel cashDividendModel = Spot);
```

Comment:
Do you think more likely that we want to pass the dividend model or the local volatility?  If the former, having it as the last argument is awkward.  In that case, we might add another pair of constructors where the `cashDividendModel` argument is before `localVol` (or even before `schemeDesc`) and deprecate the existing ones.  Thoughts?

Reply:
What do you think about a similar construct to MakeMCEuropeanEngine?

Reply:
That would be great.